File: python/problem_template.py
"""
Advent Of Code 2016
"""

import aoc2016


def solve_pt1(input_data: str) -> int:
    # solution_pt1: ???
    return 0


def solve_pt2(input_data: str) -> int:
    # solution_pt2: ???
    return 0


if __name__ == "__main__":
    tests = [
        aoc2016.AocTestCase(func=solve_pt1, output=5, input_data="abc"),
        aoc2016.AocTestCase(func=solve_pt2, output=4, input_data="def"),
    ]
    aoc2016.parser(solve_pt1, solve_pt2, tests)

File: rust/src/problem_template.rs
/* Advent Of Code 2016
 * {url}
 *
{description}
*/

use aoc2016::load_input;
use std::error::Error;

fn solve_pt1(input_text: &str) -> u64 {
    0
}

fn solve_pt2(input_text: &str) -> u64 {
    1
}

fn main() -> Result<(), Box<dyn Error>> {
    const FILENAME: &str = "../data/day_{day}_input.txt";
    let input_text = load_input(FILENAME);

    println!("Part one: {:#?}", solve_pt1(&input_text));
    // solution_pt1: ???

    println!("Part two: {:#?}", solve_pt2(&input_text));
    // solution_pt2: ???

    Ok(())
}

// Example tests
#[cfg(test)]
mod example {
    use aoc2016::test_solution;

    test_solution!(test1, solve_pt1, 5, "Your data here");
    test_solution!(test2, solve_pt2, 4, "Another data");
}
